### PR TITLE
[api] add new integration tests for API project endpoint

### LINKED ISSUE #8617: [API] Project endpoints missing 2 integration tests
2 Project endpoints do not have integration tests:
1. `/projects/{project}/instruments/{instrument}`
2. `/projects/{project}/recordings`

### PR DESCRIPTION
**Pull request by regisoc**
## Brief summary of changes

Add two endpoint integration tests:

- `/projects/{project}/instruments/{instrument}`
- `/projects/{project}/recordings`

#### Link(s) to related issue(s)

* Resolves #8617

### CODE REVIEW

File: raisinbread/test/api/LorisApiProjectsTest.php [hunk 1]
```diff
@@ -20,6 +20,8 @@ class LorisApiProjectsTest extends LorisApiAuthenticatedTest
 {
     protected $projectName = "Pumpernickel";
 
+    protected $instrumentName = "testtest";
```
Comment (kongtiaowang) [inline raisinbread/test/api/LorisApiProjectsTest.php:23]: @regisoc please using "aosi" in the test case, or keep the markTestSkipped in this  test case. I will figure out the missing bmi in docker later.

**Comment by kongtiaowang** [comment]
@regisoc I have added the bmi into testing docker.
add these changes into your PR from https://github.com/aces/Loris/pull/8650.

File: raisinbread/test/api/LorisApiProjectsTest.php [hunk 2]
```diff
@@ -519,6 +521,332 @@ public function testGetProjectsProjectInstruments(): void
...
+            ]
+        );
+        $this->assertEquals(200, $response->getStatusCode());
+        // Verify the endpoint has a body
+        $body = $response->getBody();
+        $this->assertNotEmpty($body);
+
+        $instrumentProjectArray = json_decode(
+            (string) utf8_encode(
+                $response->getBody()->getContents()
```
Comment (kongtiaowang) [inline raisinbread/test/api/LorisApiProjectsTest.php:539]:
```suggestion
                                strstr(response->getBody()->getContent(),"{")
```

File: raisinbread/test/api/LorisApiProjectsTest.php [hunk 3]
```diff
@@ -20,6 +20,8 @@ class LorisApiProjectsTest extends LorisApiAuthenticatedTest
 {
     protected $projectName = "Pumpernickel";
 
+    protected $instrumentName = "aosi";
```
Comment (kongtiaowang) [inline raisinbread/test/api/LorisApiProjectsTest.php:23]:
```suggestion
    protected $instrumentName = "bmi";
```

**Comment by kongtiaowang** [comment]
instrumentProjectArray will return "
    [Meta] => Array
        (
            [InstrumentVersion] => 1l
            [InstrumentFormatVersion] => v0.0.1a-dev
            [ShortName] => bmi
            [LongName] => BMI Calculator
            [IncludeMetaDataFields] => true
        )

    [Elements] => Array
        (
            [0] => Array
                (
                    [Type] => ElementGroup
                    [GroupType] => Page
                    [Description] => Top
                    [Elements] => Array
                        (
                            [0] => Array
                                (
                                    [Type] => date
                                    [Name] => Date_taken
                                    [Description] => Date of Administration
                                    [Options] => Array
                                        (
                                            [MinDate] => 2006-01-01
                                            [MaxDate] => 2012-12-31
                                        )

                                )

                            [1] => Array
                                (
                                    [Type] => select
                                    [Name] => Examiner
                                    [Description] => Examiner
                                    [Options] => Array
                                        (
                                            [Values] => Array
                                                (
                                                    [] => 
                                                )

                                            [AllowMultiple] => 
                                            [RequireResponse] => 
                                        )

                                )

                            [2] => Array
                                (
                                    [Type] => select
                                    [Name] => unit_classification
                                    [Description] => Choose unit of measurement
                                    [Options] => Array
                                        (
                                            [Values] => Array
                                                (
                                                    [] => 
                                                    [metric] => Metric
                                                    [imperial] => Imperial
                                                )

                                            [AllowMultiple] => 
                                            [RequireResponse] => 
                                        )

                                )

                        )

                )

            [1] => Array
                (
                    [Type] => ElementGroup
                    [GroupType] => Page
                    [Description] => Imperial units
                    [Elements] => Array
                        (
                            [0] => Array
                                (
                                    [Type] => numeric
                                    [Name] => height_feet
                                    [Description] => Your height (feet) :
                                    [Options] => Array
                                        (
                                            [NumberType] => integer
                                            [MinValue] => 
                                            [MaxValue] => 0
                                            [RequireResponse] => 1
                                        )

                                )

                            [1] => Array
                                (
                                    [Type] => numeric
                                    [Name] => height_inches
                                    [Description] => Your height (inches) :
                                    [Options] => Array
                                        (
                                            [NumberType] => integer
                                            [MinValue] => 
                                            [MaxValue] => 0
                                            [RequireResponse] => 1
                                        )

                                )

                            [2] => Array
                                (
                                    [Type] => numeric
                                    [Name] => weight_lbs
                                    [Description] => Your weight (lbs) :
                                    [Options] => Array
                                        (
                                            [NumberType] => integer
                                            [MinValue] => 50
                                            [MaxValue] => 500
                                            [RequireResponse] => 1
                                        )

                                )

                        )

                )

            [2] => Array
                (
                    [Type] => ElementGroup
                    [GroupType] => Page
                    [Description] => Metric Units
                    [Elements] => Array
                        (
                            [0] => Array
                                (
                                    [Type] => numeric
                                    [Name] => height_cms
                                    [Description] => Your height (cms) :
                                    [Options] => Array
                                        (
                                            [NumberType] => integer
                                            [MinValue] => 
                                            [MaxValue] => 0
                                            [RequireResponse] => 1
                                        )

                                )

                            [1] => Array
                                (
                                    [Type] => numeric
                                    [Name] => weight_kgs
                                    [Description] => Your weight (kgs) :
                                    [Options] => Array
                                        (
                                            [NumberType] => integer
                                            [MinValue] => 
                                            [MaxValue] => 0
                                            [RequireResponse] => 1
                                        )

                                )

                        )

                )

            [3] => Array
                (
                    [Type] => ElementGroup
                    [GroupType] => Page
                    [Description] => Results
                    [Elements] => Array
                        (
                        )

                )

        )

)
"

**Comment by regisoc** [comment]
@kongtiaowang Hi Shen, it seems to work now! At least for the instrument endpoint.
Is the same thing happening with the data for the `/projects/{project}/recordings` endpoint?
This last endpoint is working on my vm through the api_docs, not sure if it is the same issue
image

File: raisinbread/test/api/LorisApiProjectsTest.php [hunk 4]
```diff
@@ -519,6 +521,335 @@ public function testGetProjectsProjectInstruments(): void
...
+    /**
+     * Tests the HTTP GET request for the endpoint /projects/{project}/recordings
+     *
+     * @return void
+     */
+    public function testGetProjectsRecordings(): void
+    {
+        $response = $this->client->request(
+            'GET',
+            "projects/$this->projectName",
```
Comment (kongtiaowang) [inline raisinbread/test/api/LorisApiProjectsTest.php:743]:
```suggestion
            "projects/$this->projectName/recordings",
```

**Review by kongtiaowang — CHANGES_REQUESTED** [review]

**Review by kongtiaowang — APPROVED** [review]
LGTM

**Pull request merged**